girder/utility/filesystem_assetstore_adapter.py

```diff
@@ -303,7 +315,12 @@ def deleteFile(self, file):
         if matching.count(True) == 1:
             path = os.path.join(self.assetstore['root'], file['path'])
             if os.path.isfile(path):
-                os.remove(path)
+                with self.deleteLock:
+                    if self.model('upload').findOne(q) is None:
```

Comment:
We can add an index, but is it necessary?  Usually uploads are relatively transient, so we don't typically have a huge number in the system at a time.  Obviously, you can create a scenario where this is otherwise.